lib/LocationManager.php

```diff
@@ -12,17 +12,27 @@ class LocationManager {
 	public static function get_locations( $caller = false ) {
 		//prioirty: user locations, caller (but not theme), child theme, parent theme, caller
 		$locs = array();
-		$locs = array_merge($locs, self::get_locations_user());
-		$locs = array_merge($locs, self::get_locations_caller($caller));
+		$locs = array_merge_recursive( $locs, self::get_locations_user() );
+		$locs = array_merge_recursive( $locs, self::get_locations_caller( $caller ) );
 		//remove themes from caller
-		$locs = array_diff($locs, self::get_locations_theme());
-		$locs = array_merge($locs, self::get_locations_theme());
-		$locs = array_merge($locs, self::get_locations_caller($caller));
-		$locs = array_unique($locs);
+		//$locs = array_diff_assoc( $locs, self::get_locations_theme() );
```

Comment:
Will remove this.

Reply:
Just to explain, I don't think the diff is needed, it's handled in the array merge.